vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -81,31 +82,21 @@ public ITextBlock createTextBlock(String slideId, String text, Integer contentOr
         ITextBlock textBlock = textBlockFactory.createTextBlock(slide, text);
         textBlock.setContentOrder(contentOrder);
         textBlock = textBlockRepo.save((TextBlock) textBlock);
-        
         return textBlock;
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see edu.asu.diging.vspace.core.services.impl.IContentBlockManager#
-     * createImageBlock(java.lang.String, java.util.Arrays, java.lang.String)
-     */
-    @Override
-    public CreationReturnValue createImageBlock(String slideId, byte[] image, String filename, Integer contentOrder) throws ImageCouldNotBeStoredException {
-        IVSImage slideContentImage = null;
-        ISlide slide = slideManager.getSlide(slideId);
+    private IVSImage saveImage(byte[] image, IVSImage slideContentImage, String filename) {
```

Comment:
```slideContentImage``` always seems to be null and even if it weren't you're replacing whatever it is in line 92. That doesn't make sense, why do you pass it at all? Just create it and return.

Reply:
I am not sure I follow you about about slideContentImage always being null. Do you mean when it is passed into the function? I passed it in so that I did not need to declare slideContentImage in the both the function scope and the outer scope.